api/nakadi-event-bus-api.yaml

```diff
@@ -1647,6 +1695,17 @@ definitions:
         type: string
         description: |
           Offset of the event being pointed to.
+      shift:
+        type: number
+        description: |
+          This number is a modifier for the offset. It moves the cursor forward or backwards by the amount provided.
+          For example, suppose a user wants to read events starting 100 positions before "001-000D-0000000000000009A8",
+          it's possible to specify shift with -100 and Nakadi will make the necessary calculations to move the cursor
+          relatively to the given offset.
```

Comment:
I don't think we need to provide any special status code since we already have 422 for this scenario. In practice, the error message will be slightly different, to make it clearer that the requested "shifted offset" is not available. But this is not new, since we already return 422 for offsets "out of bounds".